src/VirtoCommerce.CustomerModule.Web/Controllers/Api/CustomerModuleController.cs

```diff
@@ -1,582 +1,668 @@
+                }
+                member.Addresses = addresses.ToList();
+                await _memberService.SaveChangesAsync(new[] { member });
+            }
+            return NoContent();
+        }
+
+        /// <summary>
+        /// Create employee
+        /// </summary>
+        [HttpPost]
+        [Route("employees")]
+        [Authorize(ModuleConstants.Security.Permissions.Create)]
+        public async Task<ActionResult<Employee>> CreateEmployee([FromBody] Employee employee)
+        {
+            await _memberService.SaveChangesAsync(new[] { employee });
+            return Ok(employee);
+        }
+
+        /// <summary>
+        /// Create employee
+        /// </summary>
+        [HttpPost]
+        [Route("employees/bulk")]
+        [Authorize(ModuleConstants.Security.Permissions.Create)]
+        public async Task<ActionResult<Employee[]>> BulkCreateEmployees([FromBody] Employee[] employees)
+        {
+            await _memberService.SaveChangesAsync(employees);
+            return Ok(employees);
+        }
+
+        /// <summary>
+        /// Get plenty employees 
+        /// </summary>
+        /// <param name="ids">contact IDs</param>
+        [HttpGet]
+        [Route("employees")]
+        [Authorize(ModuleConstants.Security.Permissions.Read)]
+        public async Task<ActionResult<Employee[]>> GetEmployeesByIds([FromQuery] string[] ids)
+        {
+            var result = await _memberService.GetByIdsAsync(ids, null, new[] { typeof(Employee).Name });
+            return Ok(result);
+        }
+
+        /// <summary>
+        /// Get all member organizations
+        /// </summary>
+        /// <param name="id">member Id</param>
+        [HttpGet]
+        [Route("members/{id}/organizations")]
+        public async Task<ActionResult<Organization[]>> GetMemberOrganizations([FromQuery] string id)
+        {
+            var members = await _memberService.GetByIdsAsync(new[] { id }, null, new[] { typeof(Employee).Name, typeof(Contact).Name });
+            var member = members.FirstOrDefault();
+            var organizationsIds = new List<string>();
+            if (member != null)
+            {
+                if (member is Contact contact)
+                {
+                    organizationsIds = contact.Organizations?.ToList() ?? organizationsIds;
+                }
+                else if (member is Employee employee)
+                {
+                    organizationsIds = employee.Organizations?.ToList() ?? organizationsIds;
+                }
+            }
+            var organizations = await GetOrganizationsByIds(organizationsIds.ToArray());
+            if (!(await AuthorizeAsync(organizations, ModuleConstants.Security.Permissions.Read)).Succeeded)
+            {
+                return Unauthorized();
+            }
+            return organizations;
+        }
+        #endregion
+
+        private async Task<AuthorizationResult> AuthorizeAsync(object resource, string permission)

```

Comment:
Almost all calls to `AuthorizeAsync` uses it in the same way:
```
            if (!(await AuthorizeAsync(<resource>, <permission>)).Succeeded)
            {
                return Unauthorized();
            }
```
If you extract the method, to avoid duplication maybe it would be better to include this code into it and call it e.g. `Task AuthorizeAsync(…)` .

Reply:
I can't do that because I need to return UnauthorizedResult. I may throw HttpException with 401 error code, but not sure it's correct and desired behavior

Reply:
Indeed, an additional result class would be required in that case. Throwing an exception is not a good option. Let's leave it that way.